design/ShaderLanguageFAQ.md

```diff
@@ -0,0 +1,158 @@
+# FAQ: WebGPU direction on shader language
+
+## What is the shader language for WebGPU?
+
+WebGPU will use a new shader language “WebGPU Shader Language”,
+or “WGSL” for short. This is a shading language which is designed specifically
+for, and in tandem with, WebGPU.
+
+It is under development, but will have the following traits:
+
+*   It is a text-based language.
+*   It is suitable for humans to read and write.
+*   It is strongly typed.
+*   It directly represents block-structured programming, including
+    sequence, selection (if/else), and repetition (looping).
+*   It can be faithfully and simply converted to and from SPIR-V
+    which only uses features available in WebGPU.
+*   It can be faithfully and simply converted to Metal Shading Language
+    which only uses features available in WebGPU.
+*   It can be faithfully and simply converted to HLSL
+    which only uses features available in WebGPU.
+
+The W3C WebGPU community group agreed to this direction during the
+February 2020 face-to-face meeting in Redmond.
+
+
+## How do you pronounce “WGSL”?
+
+The W3C WebGPU Community Group does not prescribe a particular
+pronunciation of “WGSL”.
+
+You can say it letter-by-letter, like “W G S L” or you can say it as a single word, like “wiggzle”, “wiggles”, or “wigsel”.
+
+
+## What does a WGSL shader look like?
+
+WGSL has not yet been specified.
+However, early prototyping work indicates the language can have a look
+and feel as represented in the following examples:
+
+A simple fragment shader:
+
+
+```
+[[location 0]] var<out> gl_FragColor : vec4<f32>;
+
+fn main() -> void {
```

Comment:
This example is out of date because of how it specifies the entry point, and the attribute syntax.

Off the top of my head:
```
[[location(0)]] var<out> gl_FragColor : vec4<f32>; 
[[stage(fragment)]]
fn main() -> void {
  gl_FragColor = vec4<f32>(0.4, 0.4, 0.8, 1.0);
}
```

Also, I'm fine to rename gl_FragColor to just `color` or `colour`.